Feature/656 course report add same as field to spreadsheet

Urjikoo opened the pull request:
## Describe your changes
 This PR adds a field to the course report download spread sheet. When a user clicks on `Download Course Report` button, whether the sameAs selected in the `customized view` menu or not, sameAs field appears next to notes column with sameAs information.  

## Types of changes
<!--
  What types of changes does your code introduce? Put an `x` in all the boxes
  that apply:
-->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality
      to change)

## Checklist:
<!--
  Go over all the following points, and put an `x` in all the boxes that apply.
-->
- [X] I have run `eslint` on the code
- [ ] I have added JSDoc for all of my code (where applicable)
- [ ] I have added tests to cover my changes.

## Priority:
- [X] Normal <!-- New piece of functionality -->
- [ ] High <!-- Critical bug requiring urgent review -->

## Related Issues:
<!--
  Add the issue number in below that this PR is intended to address.

  Also mention any issues that this PR is related to, and if possible,
  provide more information regarding the relationship to the issues in
  question (i.e does the PR fix the issue, is it designed to fix another bug
  that is similar to the issue etc.)
-->
Fixes #656 

<!--
  Attribution:
  PR Template adapted from: https://github.com/h5bp/html5-boilerplate/blob/master/.github/PULL_REQUEST_TEMPLATE.md
-->

Urjikoo committed:
Addidng sameAs fieled in table column and table row
to add sameAs field to the  downlad spreadsheet, both table header on tablecolumn and row are added.

Urjikoo committed:
Moving same as information from notes to sameAs field
currently sameAs field exists in the database but the entire field/column is empty and the course same as data is showed in notes field
along with other course information. Therefore moved the sameAs data from notes using startsWith function.

Urjikoo committed:
Removing sameAs information from notes field once it's copied to sameAs field to avoid duplicate data
removing  sameAs information from notes field to avoid duplicate data on spreadsheet.

Urjikoo committed:
Removing forloop that was moving sameAs information from notes field.
instead of moving sameAs information from notes to sameAs field, we're now getting the data from the database sameAs field.

Urjikoo committed:
Removing console log along with extra space.
removed console log where it was printing sql queries but no longer needed

natalynyu commented on src/server/report/report.service.ts:
```diff
@@ -82,6 +83,7 @@ export class ReportService {
         isSEAS: isSEASEnumToString(course.isSEAS),
         isUndergrad: course.isUndergraduate ? 'Undergraduate' : 'Graduate',
         notes: course.notes,
+        sameAs: course.sameAs,
```
The course's `sameAs` field value refers to a course's parent (if one exists). If you take a look at the `Course` entity, the `sameAs` field points to one course. From the description of the ticket, I think that we want to also pull in the course's siblings to the excel report if any exist. The logic to get this information is a bit more complicated. 

I think one way to accomplish this would be to create a separate helper function that returns an array of objects. Each key in the object would be a parent course, and the value would be an array of its the parent course's children. From there you could parse this result to get all of the `sameAs` course information. 

There are likely other ways to tackle this problem, but that was just the first idea that came to mind.

natalynyu replied:
Sorry for the confusion - you can disregard the above comment. The sameAs courses are showing as expected.

natalynyu reviewed (changes requested):
Hi Urji, you're on the right track with this, but we need some additional information to complete the goal of the ticket. The report should list not only the course parent but also any course siblings if they exist. Also, if the course itself is the parent, the `sameAs` excel field should list the course child/children.

natalynyu commented on src/server/report/report.service.ts:
```diff
@@ -51,6 +51,7 @@ export class ReportService {
       { header: 'Is Seas?', key: 'isSEAS' },
       { header: 'Is Undergrad?', key: 'isUndergrad' },
       { header: 'Notes', key: 'notes' },
+      { header: 'SameAs', key: 'sameAs' },
```
The header should probably be "Same As" for the excel sheet.

natalynyu commented on src/server/report/report.service.ts:
```diff
@@ -82,6 +83,7 @@ export class ReportService {
         isSEAS: isSEASEnumToString(course.isSEAS),
         isUndergrad: course.isUndergraduate ? 'Undergraduate' : 'Graduate',
         notes: course.notes,
+        sameAs: course.sameAs,
```
The ticket suggests possibly bolding or putting brackets around the parent course. Was this discussed further?

Urjikoo replied:
Per our discussion, leaving the font style  as is for now.

Urjikoo committed:
Adding space on report header for Same As field
to make it more readable, added space to table header on course report download sheet to the Same As field.

natalynyu reviewed (approved):
LGTM! Merge away when ready.
